### ISSUE TITLE
[netatmo] Support for switching video surveillance on/off

### ISSUE THREAD

**Issue by Novanic** [openhab/openhab-addons#7938]
It would be neat if the video surveillance of Netatmo cameras could get switched on/off with the Netatmo binding. This should be usable for all cameras (Presence and Welcome).

For example with a rule the video surveillance could get switched off automatically when someone is at home and switched on at night or when someone is leaving the house.

**Comment by Novanic** [comment]
I will try to solve that by myself.

But I have only 1 Presence camera and it should also work for the Welcome camera. Is there someone who could test that with the Welcome camera when the change is ready for testing?

**Comment by lolodomo** [comment]
Of course, I own a welcome camera.

**Comment by Novanic** [comment]
A working version can now get found here: https://github.com/Novanic/openhab-addons/tree/7938-Support-Switching-Video-Surveillance

It took me only about 1 hour to get this done after #7912 got merged. :-)

**Comment by nokyyz** [comment]
Hi @Novanic, does your implementation support the state feedback on this video surveillance channel? E.g: the surveillance state switch automatically when I switch the video surveillance on my Netatmo app (webapp or smartphone)

**Comment by Novanic** [comment]
> Hi @Novanic, does your implementation support the state feedback on this video surveillance channel? E.g: the surveillance state switch automatically when I switch the video surveillance on my Netatmo app (webapp or smartphone)

Hi, only with a delay of upto a few minutes. All data/channels are updated in an interval of about 5 minutes.

**Comment by nokyyz** [comment]
> All data/channels are updated in an interval of about 5 minutes.
Damn. 
Is it possible to configure this interval value in the binding parameters?

**Comment by lolodomo** [comment]
I guess the existing configuration setting is used. Its default value is 300000 ms (5 min) but you can adjust it.

**Comment by Novanic** [comment]
> > All data/channels are updated in an interval of about 5 minutes.
> 
> Damn.
> Is it possible to configure this interval value in the binding parameters?

For which use-case do you need an immediate update of the state?

**Comment by lolodomo** [comment]
I just tested and it works as expected with the Welcome camera. Great.
There is a little delay but we have a simlilar delay when using the netatmo app.
Maybe you should reschedule the refresh few seconds after handling this action, rather than waiting for the already scheduled refresh which can be in 5 minutes in the worst case. This will allow having uptodate channels just after switching on the surveillance.

**Comment by lolodomo** [comment]
Regarding the interval between refresh, I can confirm that we have a configuration setting for that. In case, a user wants a quickest refresh, he can reduce this interval.

In my opinion, it makes no sense to have defined this setting in milliseconds !!! We can of course expect problems if a user define 1 ms for example. @clinique : if we change now the unit, this will not be backword compatible. Maybe we should add a control to avoid a very small value. Netatmo will probably automatically banish a user in case there are too many API calls ?

**Comment by clinique** [comment]
@lolodomo : rescheduling a refresh is the best way, it's done this way in the NATherm1handler. I fully agree that millisecond is inappropriate as a refresh time scale. We could keep it as his and maybe schedule a dedicated PR for this kind of change when move to OH3 is started.

**Comment by lolodomo** [comment]
Yes, my remark about the refresh interval unit is out of the scope of this enhancement. I will create a separate PR to add a control in `defineRefreshInterval` and avoid too much small values.

In the scope of this enhancmeent is the short delay to refresh channels after activating/deactivating the surveillance.

**Comment by Novanic** [comment]
@lolodomo Thank you for testing. :-)

I don't understand why a refresh should be scheduled after switching the state within OpenHAB. When the user switches the state within OpenHAB, the state is directly updated (a switch within HabPanel for example shows directly the new value). Which positive effect should it have?

**Comment by lolodomo** [comment]
A refresh of the last event and recent live picture in particular.

**Comment by lolodomo** [comment]
And of course an update of the status. In case the item is with `autoupdate="false"`, the camera status will be really ON only after the refresh cycle.

**Comment by Novanic** [comment]
Hm, when the user switches the state from off to on within OpenHAB, the state in OpenHAB is still off (when autoupdate=false)? Could you please short confirm or explain it?

I understand that it is useful for the new floodlight switch for example. When the user switches the floodlight on, the user may want to get a new live picture where the floodlight is on (the picture has more brightness). So I can/will realize it. How many seconds should get waited before the refresh is executed? And should that be configurable?

**Comment by clinique** [comment]
Yes, you can not just rely on supposed state of the switch, that may be valid if autoupdate=true but surely false if autoupdate=false. Also, if for any reason the api post doesn't succeed, your switch will be ON while the floodlight is OFF, so re-triggering a refresh will help having a state representation as consistent as possible in regard of reality. It is also valid if the action has an effect on multiple channels (like it is for NATherm1Handler).

**Comment by lolodomo** [comment]
IMHO, no need for a configuration setting for that. You should check again because the action takes a little time even in the netatmo app. I compute around 2 seconds in the netatmo app but it looks like it takes more time when done through openHAB. But maybe it is due to how is implemented the netatmo app. The solution will be to check that the camera status is always ok with the delay you will choose.

When autoupdate is set to false, the channel state is not automatically updated by the core framework when a command is received. In general, the binding is updating the channel states. I will check and confirm that with your case. Currently my item is linked without setting autoupdate and I can see that the channel state is updated correctly.

**Comment by lolodomo** [comment]
Good point Gael, it is also a way to be sure that the command succeeded and to have a quick feedback.

**Comment by lolodomo** [comment]
@clinique : this  is doable simply by calling the method `invalidateParentCacheAndRefresh` ?

**Comment by clinique** [comment]
Yes, IIRW it's the purpose of this method

**Comment by Novanic** [comment]
Ok, the refresh is now executed and I see these positive effects:
- When the floodlight is switched on/off a new live picture gets received with the new floodlight state.
- When the video surveillance is switched on/off an event is received to confirm the new state.

I have now created a pull-request: https://github.com/openhab/openhab-addons/pull/7968
